port_drayage_plugin/src/port_drayage_worker.cpp

```diff
@@ -91,5 +91,61 @@ namespace port_drayage_plugin
         return msg;
     }
 
+    void PortDrayageWorker::on_inbound_mobility_operation(const cav_msgs::MobilityOperationConstPtr& mobility_operation_msg)
+    {
+        // Check if the received message is a new message for port drayage
+        if((mobility_operation_msg->strategy == PORT_DRAYAGE_STRATEGY_ID) && (mobility_operation_msg->strategy_params != _previous_strategy_params))
+        {
+            // Use Boost Property Tree to parse JSON-encoded strategy_params field in MobilityOperations message
+            using boost::property_tree::ptree;
+            ptree pt;
+            std::istringstream strategy_params_ss(mobility_operation_msg->strategy_params);
+            boost::property_tree::json_parser::read_json(strategy_params_ss, pt);
+            std::string received_cmv_id = pt.get<std::string>("cmv_id");
+
+            // Check if the received MobilityOperation message is intended for this vehicle's cmv_id   
+            if(received_cmv_id == _cmv_id)
+            {
+                ROS_DEBUG_STREAM("Processing new port drayage MobilityOperation message for cmv_id " << received_cmv_id);
+                _pdsm.process_event(PortDrayageEvent::RECEIVED_NEW_DESTINATION);
+                mobility_operation_message_parser(mobility_operation_msg->strategy_params);  
+                _previous_strategy_params = mobility_operation_msg->strategy_params;
+            }
+            else
+            {
+                ROS_DEBUG_STREAM("Ignoring received port drayage MobilityOperation message intended for cmv_id " << received_cmv_id);
+            }
+
+        }
+    }
+
+    void PortDrayageWorker::mobility_operation_message_parser(std::string mobility_operation_strategy_params)
+    {
+        // Use Boost Property Tree to parse JSON-encoded strategy_params field in MobilityOperations message
+        using boost::property_tree::ptree;
+        ptree pt;
+        std::istringstream mobility_operation_strategy_params_ss(mobility_operation_strategy_params);
+        boost::property_tree::json_parser::read_json(mobility_operation_strategy_params_ss, pt);
+
+        _received_cargo_id = pt.get<std::string>("cargo_id");
+        _received_operation = pt.get<std::string>("operation");
+        _received_cargo_flag = pt.get<bool>("cargo");
+        _received_start_long = pt.get<double>("location.longitude") / 10000000; // Convert 1/10 microdegrees to degrees
```

Comment:
Fixed. Created a struct to contain these data members and made the starting/destination values optional. I believe some additional freight user stories will be created to design the requirements for the MobilityOperations messages (which members of the schema will be populated) for each operation type, and from those it can be re-evaluated whether we make any of the other data members optional?